audio/src/decoder.rs

```diff
@@ -44,13 +54,14 @@ unsafe impl Send for AudioDecoderCallbacks {}
 unsafe impl Sync for AudioDecoderCallbacks {}
 
 pub struct AudioDecoderCallbacksBuilder {
-    eos: Option<Box<FnBox(u32) + Send + 'static>>,
+    eos: Option<Box<FnBox() + Send + 'static>>,
```

Comment:
can we still pass down channels here so we don't have to use shared mutable state on the DOM side to deal with this?

Reply:
We still need to use shared mutable state to [build](https://github.com/servo/servo/blob/master/components/script/dom/baseaudiocontext.rs#L447) the final decoded audio from the progress callbacks and to [create](https://github.com/servo/servo/blob/master/components/script/dom/baseaudiocontext.rs#L412) the final `AudioBuffer` content when receiving the `eos` callback. I think we can infer the number of channels from the final decoded audio array, so we don't need to send the channel count param through the `eos` callback.